x-pack/plugin/esql/qa/testFixtures/src/main/resources/rerank.csv-spec

```diff
@@ -0,0 +1,116 @@
+4536            | War and Peace (Signet Classics)                       | [John Hockenberry, Leo Tolstoy, Pat Conroy]
+9032            | War and Peace: A Novel (6 Volumes)                    | Tolstoy Leo
+2776            | The Devil and Other Stories (Oxford World's Classics) | Leo Tolstoy
+;
+
+
+reranker using multiple fields
+required_capability: rerank
+required_capability: match_operator_colon
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| RERANK "war and peace" ON title, author WITH test_reranker
+| KEEP book_no, title, author
+;
+
+book_no:keyword | title:text                                            | author:text 
+5327            | War and Peace                                         | Leo Tolstoy 
+9032            | War and Peace: A Novel (6 Volumes)                    | Tolstoy Leo 
+2776            | The Devil and Other Stories (Oxford World's Classics) | Leo Tolstoy 
+4536            | War and Peace (Signet Classics)                       | [John Hockenberry, Leo Tolstoy, Pat Conroy]
+;
+
+
+reranker after a limit
+required_capability: rerank
+required_capability: match_operator_colon
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| SORT _score DESC
+| LIMIT 3
+| RERANK "war and peace" ON title WITH test_reranker
+| KEEP book_no, title, author
+;
+
+book_no:keyword | title:text                         | author:text
+5327            | War and Peace                      | Leo Tolstoy
+4536            | War and Peace (Signet Classics)    | [John Hockenberry, Leo Tolstoy, Pat Conroy]
+9032            | War and Peace: A Novel (6 Volumes) | Tolstoy Leo
+;
+
+
+reranker before a limit
+required_capability: rerank
+required_capability: match_operator_colon
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| RERANK "war and peace" ON title WITH test_reranker
+| KEEP book_no, title, author
+| LIMIT 3
+;
+
+book_no:keyword | title:text                         | author:text
+5327            | War and Peace                      | Leo Tolstoy
+4536            | War and Peace (Signet Classics)    | [John Hockenberry, Leo Tolstoy, Pat Conroy]
+9032            | War and Peace: A Novel (6 Volumes) | Tolstoy Leo
+;
+
+
+reranker using another sort order
+required_capability: rerank
+required_capability: match_operator_colon
+
+FROM books
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| RERANK "war and peace" ON title WITH test_reranker
+| KEEP book_no, title, author
+| SORT author, title 
+| LIMIT 3
+;
+
+book_no:keyword | title:text                                            | author:text
+4536            | War and Peace (Signet Classics)                       | [John Hockenberry, Leo Tolstoy, Pat Conroy]
+2776            | The Devil and Other Stories (Oxford World's Classics) | Leo Tolstoy
+5327            | War and Peace                                         | Leo Tolstoy
+;
+
+
+reranker after RRF
```

Comment:
I think this test we can remove - it will never work on 8.19 since `rrf` nor `fork` should be available